src/lib.rs

```diff
@@ -12,14 +12,179 @@
 
 /// This module exports a C2PA library
 use std::env;
+use std::sync::RwLock;
 
-pub(crate) use c2pa_c::{
-    read_file, read_ingredient_file, sdk_version, sign_file, Error, SignerInfo,
-};
+pub use c2pa::SigningAlg;
+
+/// these all need to be public so that the uniffi macro can see them
+mod error;
+pub use error::{Error, Result};
+#[cfg(feature = "v1")]
+mod json_api;
+#[cfg(feature = "v1")]
+pub use json_api::{read_file, read_ingredient_file, sign_file};
+#[cfg(feature = "v1")]
+mod signer_info;
+#[cfg(feature = "v1")]
+pub use signer_info::{CallbackSigner, SignerCallback, SignerConfig, SignerInfo};
```

Comment:
It looks like you are still keeping the V1 stuff in here? It might be good to put this under a `v1` namespace or something so it's clear to other people referencing the code (looking at people such as myself) which parts are code related to the new way.

Reply:
I'll strip this out. I was keeping it in to make sure I'd covered the functional equivalents.